mpy-cross/main.c

```diff
@@ -64,6 +68,28 @@ STATIC void stderr_print_strn(void *env, const char *str, size_t len) {
 
 STATIC const mp_print_t mp_stderr_print = {NULL, stderr_print_strn};
 
+STATIC void fd_print_strn(void *env, const char *str, size_t len) {
+    int fd = (intptr_t)env;
+    MP_THREAD_GIL_EXIT();
+    ssize_t ret = write(fd, str, len);
+    MP_THREAD_GIL_ENTER();
+    (void)ret;
+}
+
+void mp_raw_code_save_file(mp_compiled_module_t *cm, const char *filename) {
```

Comment:
This function could be useful for more than just mpy-cross.  Eg it could be used by the unix port if it ever supports saving .mpy files, for example like `micropython.save_mpy(...)`.  So that's why I thought it was OK to keep it in `py/persistentcode.c`, similar to how `mp_lexer_new_from_fd()` is in `py/lexer.c` and guarded by `MICROPY_HELPER_LEXER_UNIX`.